### PR TITLE
SIDM-8447 - Updated invite user to use V2 endpoints

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
### JIRA link ###

https://tools.hmcts.net/jira/SIDM-8447

### Change description ###

- Added AuthorizedAxios which auth's with client credential
- Added InviteService which makes calls to V2 invite endpoints
- Updated existing controllers/unit tests to make use of new InviteService
- Removed redundant functional test

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/app/invite-service/InviteService.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    invitedBy?: string,
+    successRedirect?: string
+  ) => {
+    return this.idamApiAxios
+      .post(this.INVITE_ENDPOINT, {
+        invitationType: invitationTypes.INVITE,
+        email,
+        forename,
+        surname,
+        clientId: this.CLIENT_ID,
```
Comment (jburke-idam) [inline src/main/app/invite-service/InviteService.ts:36]: I think this should be the client id of the service inviting the user (so, adoption for private beta)

File: src/main/controllers/AddUserRolesController.ts [hunk 2]
```diff
@@ -30,12 +37,14 @@ export class AddUserRolesController extends RootController {
...
-      lastName: fields._surname,
-      roles: convertToArray(roles)
-    });
+
+    await this.inviteService.inviteUser(
+      fields._email,
+      fields._forename,
+      fields._surname,
+      convertToArray(roles),
+      req.session.user.id
```
Comment (jburke-idam) [inline src/main/controllers/AddUserRolesController.ts:46]: you'll need to set a success redirect here or the user won't go anywhere after activation

File: src/main/app/invite-service/InviteService.ts (hunk 1 above)
Comment (Thomas-Geraghty) [reply src/main/app/invite-service/InviteService.ts]: Addressed in https://github.com/hmcts/idam-user-dashboard/pull/440/commits/e2eb3d1c2f1531add059105a5dc9196c92ecaff7

File: src/main/controllers/AddUserRolesController.ts (hunk 2 above)
Comment (Thomas-Geraghty) [reply src/main/controllers/AddUserRolesController.ts:46]: Where should we redirect users who have been invited through the dashboard? We don't know what service they're being invited for here (as it's just name, email and roles) so I'm not sure how the redirect should be used here

Comment (jburke-idam) [reply src/main/controllers/AddUserRolesController.ts:46]: I think the old endpoints would have defaulted to idam-user-dashboard being the client, so the redirect would be the dashboard one. I think it's fine to do that here - might be interesting to have a new requirement to select the "destination" service for someone being invited.

File: config/default.yaml [hunk 3]
```diff
@@ -16,13 +16,14 @@ services:
...
       endSession: '/o/endSession'
+      invite: '/api/v2/invitations'
     systemUser:
       username: 'idam-user-dashboard-system-user@user.idam'
       password: 'Pa55word11'
     clientID: 'idam_user_dashboard'
     clientSecret: 'idam_user_dashboard_secret'
     callbackURL: 'http://localhost:3100/oauth2/callback'
-    scope: 'openid profile roles manage-user search-user create-user'
+    scope: 'profile roles manage-user search-user create-user create-invitation'
```
Comment (kremi) [inline config/default.yaml:26]: I don't think users logging into User Dashboard should be given that scope (for now). 
Maybe we need separate configuration for the scopes required for the client_credentials authentication.

File: src/main/controllers/AddUserRolesController.ts (hunk 2 above)
Comment (Thomas-Geraghty) [reply src/main/controllers/AddUserRolesController.ts:46]: TODO:
- Redirect new support users to dashboard
- Redirect new professional users to gov.uk

File: config/default.yaml (hunk 3 above)
Comment (Thomas-Geraghty) [reply config/default.yaml]: Addressed in https://github.com/hmcts/idam-user-dashboard/pull/440/commits/ae420a46043915790f0454563342272b2f8f7591

File: src/main/controllers/AddUserRolesController.ts (hunk 2 above)
Comment (Thomas-Geraghty) [reply src/main/controllers/AddUserRolesController.ts]: Addressed in https://github.com/hmcts/idam-user-dashboard/pull/440/commits/a9f590b9b2832e7369095b0d410ac293fd2a4f65

File: src/main/controllers/AddUserRolesController.ts [hunk 4]
```diff
@@ -30,12 +42,29 @@ export class AddUserRolesController extends RootController {
...
+        successRedirect: serviceInfo.hmctsAccess.postActivationRedirectUrl
+      });
+    } else {
+      await this.inviteService.inviteUser({
+        email: fields._email,
+        forename: fields._forename,
+        surname: fields._surname,
+        activationRoleNames: convertToArray(roles),
+        invitedBy: req.session.user.id,
+        clientId: serviceInfo.clientId
```
Comment (kremi) [inline src/main/controllers/AddUserRolesController.ts:64]: What would be the result of not passing any `successRedirect` here?

Comment (jburke-idam) [reply src/main/controllers/AddUserRolesController.ts:64]: The activate user stuff in api returns an AcrtivationResult object which has a redirect url in it which comes from this value. If the api returns null after activation then I think web public defaults to "https://www.gov.uk/" for the continue button

File: config/default.yaml [hunk 5]
```diff
@@ -16,13 +16,16 @@ services:
...
+      invite: '/api/v2/invitations'
+      services: '/api/v2/services'
     systemUser:
       username: 'idam-user-dashboard-system-user@user.idam'
       password: 'Pa55word11'
     clientID: 'idam_user_dashboard'
     clientSecret: 'idam_user_dashboard_secret'
     callbackURL: 'http://localhost:3100/oauth2/callback'
     scope: 'openid profile roles manage-user search-user create-user'
+    clientScope: 'create-invitation view-service-provider'
```
[the remark this answers is not in the document]
Comment (Thomas-Geraghty) [reply config/default.yaml]: Addressed in https://github.com/hmcts/idam-user-dashboard/pull/440/commits/17d2119f2e6b4871b2dfe035a12d043a0c5a5e8d

**Review by kremi — APPROVED** [review]

**Pull request merged**